homeassistant/components/hive.py

```diff
@@ -0,0 +1,707 @@
+"""Hive Integration - Platform."""
+import logging
+from datetime import datetime
+from homeassistant.helpers.discovery import load_platform
+
+REQUIREMENTS = ['pyhiveapi==0.1.1']
+
+HGO = None
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = 'hive'
+
+
+class HivePlatformData:
+    """Initiate Hive PlatformData Class."""
+
+    min_max_data = {}
+
+
+class HiveSession:
+    """Initiate Hive Session Class."""
+
+    username = ""
+    password = ""
+    data = HivePlatformData()
+    logging = False
+    hass = None
+    core = None
+    heating = None
+    hotwater = None
+    light = None
+    sensor = None
+    switch = None
+
+
+def setup(hass, config):
+    """Setup the Hive platform."""
+    from pyhiveapi import Pyhiveapi
+
+    hsc_s = HiveSession()
+    hsc_s.core = Pyhiveapi()
+
+    hsc_s.hass = hass
+
+    hsc_s.username = None
+    hsc_s.password = None
+    api_mins_between_updates = 2
+    api_dl_all = {}
+    api_dl_sensor = []
+    api_dl_climate = []
+    api_dl_light = []
+    api_dl_switch = []
+
+    hive_config = config[DOMAIN]
+
+    if "username" in hive_config and "password" in hive_config:
+        hsc_s.username = config[DOMAIN]['username']
+        hsc_s.password = config[DOMAIN]['password']
+    else:
+        _LOGGER.error("Missing UserName or Password in config")
+
+    if "minutes_between_updates" in hive_config:
+        api_mins_between_updates = config[DOMAIN]['minutes_between_updates']
+    else:
+        api_mins_between_updates = 2
+
+    if "logging" in hive_config:
+        if config[DOMAIN]['logging']:
+            hsc_s.logging = True
+            _LOGGER.warning("Logging is Enabled")
+        else:
+            hsc_s.logging = False
+    else:
+        hsc_s.logging = False
+
+    if hsc_s.username is None or hsc_s.password is None:
+        _LOGGER.error("Missing UserName or Password in Hive Session details")
+    else:
+
+        api_dl_all = hsc_s.core.initialise_api(hsc_s.username,

```

Comment:
I guess that there will be an exception if the init fails. Wrong username or password.

Reply:
this section has been completely re-written